auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -885,9 +1040,17 @@ private LoadedIdentity loadIdentityPrivileged(final boolean skipCredentials, fin
                 inputFactory.setProperty(XMLInputFactory.IS_COALESCING, Boolean.TRUE);
                 try (final AutoCloseableXMLStreamReaderHolder holder = new AutoCloseableXMLStreamReaderHolder(inputFactory.createXMLStreamReader(inputStream, "UTF-8"))) {
                     final XMLStreamReader streamReader = holder.getXmlStreamReader();
-                    return parseIdentity(streamReader, skipCredentials, skipAttributes);
+                    LoadedIdentity parsedIdentity = parseIdentity(streamReader, skipCredentials, skipAttributes);
+                    if (this.publicKey != null) {
+                        createDigitalSignature(this.path.toString(), this.name);
```

Comment:
can createDigitalSignature be deleted from this method?

Reply:
Ah yes it can, will remove